src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -63,12 +63,12 @@ public Parameter[] GetOrderedParametersByRequired()
             return OrderParametersByRequired(_parameters.Values);
         }
 
-        public static IEnumerable<InputParameter> GetParametersFromOperations(IEnumerable<InputOperation> operations) =>
-            operations
-                .SelectMany(op => op.Parameters)
-                .Where(p => p.Kind == InputOperationParameterKind.Client)
-                .Distinct()
-                .ToList();
+        public static IEnumerable<InputParameter> GetParametersFromClient(InputClient client)
+        {
+            return (client.Parameters ?? new List<InputParameter>())
+                .Concat(client.Operations.SelectMany(op => op.Parameters).Where(p => p.Kind == InputOperationParameterKind.Client))
+                .Distinct();
```

Comment:
is it possible that two operations have parameters with the same name and kind, but they are different instances?
This could come from an issue in our emitter (for example they should be the same instance, but we incorrectly made them different) or come from a case they just have different parameter but by accident with same name and kind (different type for example)
If this could happen, here we may have trouble.

In summary that I sense this piece of code is fragile if we could find a way to make it more solid, it would be better.

Reply:
Current in out tspCodeModel, those client parameter in operations is the same instance (using ref to connect), so in most case it is not cause problem. 

But yes, we may need to find a way to make it more solid, especially when we adopt client.parameters from tcgc

Reply:
You are saying 
```
op method1(@global parameter: string): void;
op method2(@global parameter: string): void;
```
Customer wants these two parameters to be different instances. Then there must a way to let us/emitter know (no matter by decorators or other ways). We get this information and deal it with different `Parameter`s. 

I don't worry too much about it:
1. Customer writes bad spec, they should take responsibility themselves.
2. The same logic exists in M4.
3. Maybe in the new InputModel, the global parameters is a property of `InputClient` instead of `InputOperation`.

Reply:
If we follow option 3, I agree there is no issues.

For 1, yes the customer should take responsibility if they wrote bad specs and bad API was generated. But if we generate compilation errors, they will just claim there is a bug in the generator.
Bad spec => bad API
but not
Bad spec /=> invalid code